packages/core/src/resolveTrustChains/policies/MetadataHelper.ts

```diff
@@ -0,0 +1,38 @@
+import type { Metadata } from '../../metadata'
+
+export class MetadataHelper {
```

Comment:
looking at this implementation it should be quite simple to avoid the `cloneDeep` and just take an object as input, and modify it based on immutability. 

```

  public setPropertyValue(service: string, property: string, value: unknown) {
    this.leafMetadata = {
      [service]: {
       ...this.leafMetadata[service], // optionally add a helper method to get a value or an empty object otherwise
      [property]: value
     }
   }
    this.leafMetadata[service] ??= {}
    this.leafMetadata[service][property] = value
  }
```

You can even use https://www.30secondsofcode.org/js/s/immutable-object-proxy to avoid accidental write on the source object.

Reply:
it does add complexity, but I think JSON.stringify/parse should be used as a last resort